### ISSUE TITLE
Add oversampling setting adjustment

### ISSUE THREAD

**Issue by jaxxzer** [bluerobotics/BlueRobotics_MS5837_Library#10]
This will allow us to reduce the blocking measurement [delay](https://github.com/bluerobotics/BlueRobotics_MS5837_Library/blob/master/MS5837.cpp#L66) time.

**Comment by jaxxzer** [comment]
For example, our python library already supports this: https://github.com/bluerobotics/ms5837-python#readoversamplingosr_8192

**Comment by senceryazici** [comment]
Absolutely, right now we're having issues using our bar30 sensor, as the sensor spends time of 40ms to read, it reduces our communication rate down to ~50Hz, 

since the oversampling can be set to 256, it can be running at 1.5Khz.
Please implement this feature to the arduino library.

Thanks

**Comment by senceryazici** [comment]
I've modified such changes on my forked repository. => https://github.com/senceryazici/BlueRobotics_MS5837_Library/tree/pr_osr_implementation

I can open a pull request if you like.

Thanks

**Comment by jaxxzer** [comment]
> I can open a pull request if you like.

@senceryazici yes please :+1:

**Comment by senceryazici** [comment]
Pending pull request #13

**Comment by kiteborne** [comment]
Why does it not compile on teensy 3.2 ? - is it some of the math in the library (POW) forexample ?
